### PR TITLE
env configurable logger

### PR DESCRIPTION
**Pull request by Pavel-Durov**
+ added env configurable logger support

Output example:
```
λ RUST_LOG=debug  cargo run -- '2+?'
    Finished dev [unoptimized + debuginfo] target(s) in 0.02s
     Running `target/debug/yaiwr '2+?'`
[2023-03-07T11:24:11Z DEBUG yaiwr] input: "2+?"
[2023-03-07T11:24:11Z DEBUG yaiwr] AST: Err("Parsing error at line 1 column 3. Repair sequences found:\n   1: Insert INT, Delete ?")
[2023-03-07T11:24:11Z ERROR yaiwr] Evaluation error: Parsing error at line 1 column 3. Repair sequences found:
       1: Insert INT, Delete ?
Evaluation error: Parsing error at line 1 column 3. Repair sequences found:
   1: Insert INT, Delete ?

```

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -27,16 +29,25 @@ fn main() {
...
-                Err(msg) => eprintln!("Evaluation error: {}", msg),
+                Err(msg) => {
+                    error!("Evaluation error: {}", msg);
+                    eprintln!("Evaluation error: {}", msg);
+                }
             }
         }
-        Err(msg) => eprintln!("Evaluation error: {}", msg),
+        Err(msg) => {
+            error!("Evaluation error: {}", msg);
```
Comment (vext01) [inline src/main.rs:49]: It's gonna print that anyway, whether or not logging is enabled.

Comment (Pavel-Durov) [reply src/main.rs:49]: yes, I thought that should be expected - this error is the user feedback.
Otherwise no feedback will be given for parsing errors

Comment (Pavel-Durov) [reply src/main.rs:49]: actually:
`error!("Evaluation error: {}", msg);`
will be printed based on the env var
however:
`eprintln!("Evaluation error: {}", msg);`
will be always printed

Comment (Pavel-Durov) [reply src/main.rs:49]: output example:
```
$ RUST_LOG=error  cargo run -- '2+?'
[2023-03-07T11:48:29Z ERROR yaiwr] Evaluation error: Parsing error at line 1 column 3. Repair sequences found:
       1: Insert INT, Delete ?
Evaluation error: Parsing error at line 1 column 3. Repair sequences found:
   1: Insert INT, Delete ?
```
vs
```
$ RUST_LOG=nada  cargo run -- '2+?'
Evaluation error: Parsing error at line 1 column 3. Repair sequences found:
   1: Insert INT, Delete ?
```

Comment (Pavel-Durov) [reply src/main.rs:48]: that's surprising, I expected the undefined `RUST_LOG` to be the same as the non-set level.
removed :point_right: https://github.com/softdevteam/pavel.yaiwr/pull/11/commits/24c4e16f7c76d1964020c351ee1db624d07be536

Comment (vext01) [reply src/main.rs:48]: Is that right? Should the error always be printed?

I was expecting you to remove the `error!`, but leave the `eprintln!`

Comment (Pavel-Durov) [reply src/main.rs]: I don't see that much difference both will print an error message to stderr.
But, maybe having non-structured logs as user feedback makes more sense.  I set `eprintln!` back :point_right: https://github.com/softdevteam/pavel.yaiwr/pull/11/commits/3c83989b46339beb0bc588f93ae82958d1a8677e

Comment (vext01) [reply src/main.rs]: That makes more sense to me, as the other way around the user will see no error message at all if logging isn't enabled.

**Comment by vext01** [comment]
please squash

**Comment by Pavel-Durov** [comment]
squashed :point_right: https://github.com/softdevteam/pavel.yaiwr/pull/11/commits/6ca86a94a0475545d2cc3f0b1753ce7490ee638a

**Comment by Pavel-Durov** [comment]
failed on `cargo fmt --all -- --check`, I'm pushing fix in a bit

**Pull request merged**